src/ozone/ozone.py

```diff
@@ -520,6 +521,75 @@ def get_specific_parameter(
 
         return result
 
+    def get_search_results_historical(self, city: str):
+        r = requests.get(f"https://search.waqi.info/nsearch/station/{city}")
+        res = r.json()
+
+        city_id, country_code, station_name, city_url, score = [], [], [], [], []
+
+        for candidate in res["results"]:
+            city_id.append(candidate["x"])
+            country_code.append(candidate["c"])
+            station_name.append(candidate["n"])
+            city_url.append(candidate["s"].get("u"))
+            score.append(candidate["score"])
+
+        return pandas.DataFrame(
+            {
+                "city_id": city_id,
+                "country_code": country_code,
+                "station_name": station_name,
+                "city_url": city_url,
+                "score": score,
+            }
+        ).sort_values(by=["score"], ascending=False)
+
+    @functools.lru_cache(maxsize=128)
+    def get_historical_data(
```

Comment:
This method is a little intimidating to use, as at least on my computer when I hover over it with my mouse, it doesn't show me any useful information (docstring, argument types and names) in the popup, but instead only shows a line about it being a _lru_cache_wrapper. 

While the user could guess, or use some common sense to enter the arguments, a visible docstring is always helpful.
<img width="810" alt="Screenshot 2022-04-12 at 12 22 05 AM" src="https://user-images.githubusercontent.com/68847270/162786551-ef7ced6b-293b-411f-8554-2d4f8c875115.png">

<img width="602" alt="Screenshot 2022-04-12 at 12 22 41 AM" src="https://user-images.githubusercontent.com/68847270/162786544-b1dfeb0b-dab1-4001-8af3-0ba7138c4d08.png">

I've attached images to clarify what I mean. One depicts normal behaviour, the other how this method behaves.

Reply:
> I understand the problem you demonstrated. This was not obvious to me. It's definitely because the method is wrapped by a decorator function. (i.e. the decorator replaces the method including type hint and docstring)
> 
> I've googled but apparently the easiest way to make it work again is just to remove the decorator. Other ways exist -- e.g. using `functools.wrap` and write a custom decorator; or move the caching to some internal method with no visible docstring to the user -- but I don't think it's worth the hassle for now. I will research more and probably add it back in some other time.

Yeahhh I thought as much. I don't think removing the decorator or any similar solution is worth the extra work it will create. We can certainly add something like this in a patch later.

For now we can go with good official API documentation, and maybe add an example to the README for clarity.